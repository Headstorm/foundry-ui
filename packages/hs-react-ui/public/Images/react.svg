<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 19.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_2" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 width="561.8px" height="499.9px" viewBox="-80.4 151.2 561.8 499.9" style="enable-background:new -80.4 151.2 561.8 499.9;"
	 xml:space="preserve">
<style type="text/css">
	.st0{fill:#939598;}
</style>
<circle class="st0" cx="200.5" cy="400.6" r="50.2"/>
<path class="st0" d="M200.5,515.6c-70.5,0-132.1-8.3-178.2-24.1c-29.9-10.2-55.3-23.8-73.4-39.3c-19.2-16.4-29.4-34.3-29.4-51.6
	c0-33.2,36.4-65.7,97.5-86.9c50-17.4,115.2-27.1,183.4-27.1c67,0,131.3,9.4,181,26.6c29.1,10,53.6,23,71,37.4
	c18.9,15.8,28.9,33.1,28.9,50c0,34.5-40.7,69.4-106.3,91.1C328.6,507.1,266.6,515.6,200.5,515.6z M200.5,310.6
	c-64.7,0-128.7,9.4-175.5,25.7c-40.5,14.1-81.4,39.5-81.4,64.3c0,10,7.4,21.8,21,33.4c15.8,13.5,38.4,25.6,65.5,34.8
	c43.6,14.9,102.6,22.8,170.4,22.8c63.6,0,122.9-8,167-22.7c61.7-20.5,89.9-49.8,89.9-68.3c0-9.5-7.2-20.7-20.3-31.6
	c-15.1-12.6-37.1-24.1-63.4-33.2C326.4,319.6,264.9,310.6,200.5,310.6z"/>
<path class="st0" d="M86.6,650.8c-10.2,0-19.2-2.2-26.8-6.6C31,627.6,21.1,579.8,33.1,516.3c9.9-52.1,34.1-113.3,68.2-172.4h0
	c33.5-58,73.7-109,113.4-143.5c23.2-20.2,46.7-35,67.9-42.8c23.1-8.5,43.1-8.5,57.7-0.1c29.9,17.2,39.8,70,25.8,137.6
	c-9.9,48-33.5,105.9-66.5,163.2c-35.2,61-73.2,110.2-109.9,142.3c-23.8,20.8-48.3,36-70.7,43.9C107.4,648.7,96.4,650.8,86.6,650.8z
	 M122.1,355.9c-32.3,56-56.2,116.1-65.4,164.9c-8,42.1-6.5,90.2,15,102.6c16.1,9.3,55,0.5,102.3-40.9c34.7-30.3,71-77.5,104.9-136.2
	c31.8-55.1,54.4-110.5,63.8-156c6-29.1,7.3-55.5,3.8-76.4c-3-17.9-9.4-30.6-18.2-35.6c-8.2-4.7-21.5-4.1-37.5,1.8
	c-18.5,6.8-39.4,20.1-60.4,38.4C192.8,251.3,154.3,300.1,122.1,355.9L122.1,355.9z"/>
<path class="st0" d="M314.4,651.1c-27.2,0-61.7-16.4-97.7-47.4c-40.2-34.6-81.1-86.1-115.3-145.2v0c-33.6-58-57.6-118.3-67.7-170
	c-5.9-30.2-7-57.9-3.2-80.2c4.2-24.3,14.1-41.6,28.8-50.1c29.8-17.3,80.5,0.5,132.1,46.4c36.6,32.5,75,81.9,108.1,139.1
	c35.3,61,59,118.5,68.4,166.3c6.1,31,7.1,59.8,2.8,83.2c-4.6,24.9-15,42.6-30,51.3C333.2,648.9,324.3,651.1,314.4,651.1z
	 M122.2,446.5c32.4,56,72.6,106.7,110.2,139c32.5,28,74.9,50.6,96.4,38.2c16.1-9.3,27.9-47.4,15.7-109
	c-9-45.2-31.7-100.2-65.6-158.9c-31.9-55.1-68.6-102.4-103.3-133.2c-22.2-19.7-44.4-34.1-64.2-41.5c-17-6.4-31.2-7.1-39.9-2.1
	c-8.2,4.7-14.3,16.6-17.2,33.4c-3.3,19.4-2.3,44.2,3.1,71.5C66.9,333,89.9,390.7,122.2,446.5L122.2,446.5z"/>
</svg>
