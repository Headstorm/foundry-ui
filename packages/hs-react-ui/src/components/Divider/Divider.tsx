import React from 'react';
import styled, { StyledComponentBase } from 'styled-components';

import Colors from '../../enums/colors';

export const DefaultDivider = styled.hr`
  ${({ width = '90%', height = '1px' }: { width: string; height: string }) => `
    border: none;
    height: ${height};
    width: ${width};
    background-color: ${Colors.grayLight};
  `}
`;

export const DefaultDividerContainer = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 10px;
  margin-bottom: 10px;
`;

export interface DividerProps {
  StyledDivider?: StyledComponentBase<any, {}>;
  StyledDividerContainer?: StyledComponentBase<any, {}>;
  width?: string;
  height?: string;
}

const Divider = ({
  StyledDivider = DefaultDivider,
  StyledDividerContainer = DefaultDividerContainer,
  width = '90%',
  height = '1px',
}: DividerProps) => (
  <StyledDividerContainer>
    <StyledDivider width={width} height={height} />
  </StyledDividerContainer>
);

Divider.Container = DefaultDividerContainer;
export default Divider;
