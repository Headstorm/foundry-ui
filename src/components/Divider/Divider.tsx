import React from 'react';
import styled from 'styled-components';

import { useSeparator } from 'react-aria';
import { Div, HR } from '../../htmlElements';
import { SubcomponentPropsType, StyledSubcomponentType } from '../commonTypes';
import { useTheme } from '../../context';

export const DefaultDivider = styled(HR)`
  ${({ width = '90%', height = '1px' }: { width: string; height: string }) => {
    const { colors } = useTheme();
    return `
      border: none;
      height: ${height};
      width: ${width};
      background-color: ${colors.grayLight};
  `;
  }}
`;

export const DefaultDividerContainer = styled(Div)`
  display: flex;
  justify-content: center;
  margin-top: 10px;
  margin-bottom: 10px;
`;

export interface DividerProps {
  StyledDivider?: StyledSubcomponentType;
  StyledDividerContainer?: StyledSubcomponentType;
  dividerProps?: SubcomponentPropsType;
  dividerContainerProps?: SubcomponentPropsType;

  width?: string;
  height?: string;

  containerRef?: React.RefObject<HTMLDivElement>;
  dividerRef?: React.RefObject<HTMLHRElement>;
}

const Divider = ({
  StyledDivider = DefaultDivider,
  StyledDividerContainer = DefaultDividerContainer,
  dividerProps = {},
  dividerContainerProps = {},
  width = '90%',
  height = '1px',
  containerRef,
  dividerRef,
}: DividerProps): JSX.Element => {
  const { separatorProps: ariaProps } = useSeparator(dividerProps);
  return (
    <StyledDividerContainer
      data-test-id="hsui-Divider"
      {...dividerContainerProps}
      ref={containerRef}
    >
      <StyledDivider
        {...ariaProps}
        width={width}
        height={height}
        ref={dividerRef}
        {...dividerProps}
      />
    </StyledDividerContainer>
  );
};

Divider.Container = DefaultDividerContainer;
Divider.Divider = DefaultDivider;
export default Divider;
