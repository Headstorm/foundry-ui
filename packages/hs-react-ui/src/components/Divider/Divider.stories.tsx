import React from 'react';
import styled from 'styled-components';
import { text } from '@storybook/addon-knobs';

import { storiesOf } from '@storybook/react';

import colors from '../../enums/colors';
import Divider from './Divider';

storiesOf('Divider', module)
  .addParameters({ component: Divider })
  .add('Default', () => <Divider width={text('width', '10rem')} height={text('height', '1px')} />)
  .add('Themed', () => {
    const ThemedDivider = styled.hr`
      ${({ height, width }: { height: string; width: string }) => `
      box-shadow: 10px, 10px, ${colors.primary};
      border: none;
      background: radial-gradient(${colors.primary}, transparent);
      height: ${height};
        width: ${width};
    `}
    `;
    return (
      <Divider
        StyledDivider={ThemedDivider}
        width={text('width', '10rem')}
        height={text('height', '2px')}
      />
    );
  });
