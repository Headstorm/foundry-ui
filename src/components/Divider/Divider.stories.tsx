import React from 'react';
import styled from 'styled-components';

import { Story, Meta } from '@storybook/react';

import colors from '../../enums/colors';
import Divider from './Divider';

const Template: Story = args => <Divider {...args} />;

export const Default = Template.bind({});
Default.args = {
  width: '10rem',
  height: '1px',
};

const ThemedDivider = styled.hr`
  ${({ height, width }: { height: string; width: string }) => `
      box-shadow: 10px, 10px, ${colors.primary};
      border: none;
      background: radial-gradient(${colors.primary}, transparent);
      height: ${height};
        width: ${width};
    `}
`;

export const Themed = Template.bind({});
Themed.args = {
  ...Default.args,
  height: '2px',
  StyledDivider: ThemedDivider,
};

export default {
  title: 'Divider',
  component: Divider,
} as Meta;
