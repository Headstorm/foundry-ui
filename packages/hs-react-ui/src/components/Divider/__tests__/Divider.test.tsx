import React from 'react';
import { render, fireEvent, waitFor, configure } from '@testing-library/react';
import Divider from '../Divider';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);
configure({ testIdAttribute: 'data-test-id' });

const testId = 'hsui-Divider';

describe('Divider', () => {
  it('matches snapshot', async () => {
    const { container, getByTestId } = render(<Divider />);
    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with props', async () => {
    const { container, getByTestId } = render(<Divider width="50%" height="2rem" />);
    await waitFor(() => getByTestId(testId));
    expect(container).toMatchSnapshot();
  });
  describe('Accessibility Tests', () => {
    it('Should pass accessibility test with default props', async () => {
      const component = <Divider />;
      const { container } = render(component);
      const results = await axe(container);
      expect(results).toHaveNoViolations();
    });
  });
  describe('Ref Tests', () => {
    it('containerRef.current should exist', async () => {
      const ref = React.createRef<HTMLDivElement>();
      const { getByTestId } = render(<Divider containerRef={ref} testId={testId} />);
      await waitFor(() => getByTestId(testId));
      expect(ref.current instanceof HTMLDivElement).toBeTruthy();
    });
    it('dividerRef.current should exist', async () => {
      const ref = React.createRef<HTMLHRElement>();
      const { getByTestId } = render(<Divider dividerRef={ref} testId={testId} />);
      await waitFor(() => getByTestId(testId));
      expect(ref.current instanceof HTMLHRElement).toBeTruthy();
    });
  });
});
