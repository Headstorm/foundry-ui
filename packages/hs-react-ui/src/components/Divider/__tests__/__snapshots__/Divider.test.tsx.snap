// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider matches snapshot 1`] = `
.c1 {
  border: none;
  height: 1px;
  width: 90%;
  background-color: #7c8186;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 10px;
  margin-bottom: 10px;
}

<div>
  <div
    class="c0"
    data-test-id="hsui-Divider"
  >
    <hr
      class="c1"
      height="1px"
      width="90%"
    />
  </div>
</div>
`;

exports[`Divider matches snapshot with props 1`] = `
.c1 {
  border: none;
  height: 2rem;
  width: 50%;
  background-color: #7c8186;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 10px;
  margin-bottom: 10px;
}

<div>
  <div
    class="c0"
    data-test-id="hsui-Divider"
  >
    <hr
      class="c1"
      height="2rem"
      width="50%"
    />
  </div>
</div>
`;
